src/mpags-cipher.cpp

```diff
@@ -88,23 +96,18 @@ int main(int argc, char* argv[])
             inputText += transformChar(inputChar);
         }
     }
-
-    std::string outputText;
-
-    switch (settings.cipherType[0]) {
-        case CipherType::Caesar: {
-            // Run the Caesar cipher (using the specified key and encrypt/decrypt flag) on the input text
-            CaesarCipher cipher{settings.cipherKey[0]};
-            outputText = cipher.applyCipher(inputText, settings.cipherMode);
-            break;
-        }
-        case CipherType::Playfair: {
-            PlayfairCipher cipher{settings.cipherKey[0]};
-            outputText = cipher.applyCipher(inputText, settings.cipherMode);
-            break;
-        }
+    
+     std::vector<std::unique_ptr<Cipher>> ciphers;     // Store cipher instances
+    for (std::size_t i = 0; i < settings.nExpectedCiphers; ++i) {
+        ciphers.push_back(CipherFactory::makeCipher(settings.cipherType[i], settings.cipherKey[i]));   // Use cipher factory to make each cipher from cmdline
     }
 
+    // Loop through the collection of ciphers
```

Comment:
Before doing this, you ought to reverse the order of the container if you're decrypting.  The Caesar and Vigenere ciphers are commutable but Playfair definitely isn't!